### ISSUE TITLE
Enhancement - Choice of DNS over HTTPS Services

### ISSUE THREAD

**Issue by sanitybit** [jessejanderson/skylink#11]
The post-install privacy consent workflow could be reworked to allow the user to choose between the Google or [Cloudflare DoH endpoint](https://developers.cloudflare.com/1.1.1.1/encryption/dns-over-https/make-api-requests/dns-json/) (or maybe another provider). The selection would be stored in local storage.

I'll admit this may be a boondoggle, since imho, the privacy policies of all the major DoH providers are pretty similar, but if enough people complain about the Google requirement, I'd pick up the mantle on this one.

**Comment by sanitybit** [comment]
@jessejanderson @aliceisjustplaying 

I don't think this is resolved by #18; the addition of an alternate method hasn't removed the dependency on Google DNS for the primary method.

I'll put together a PR for this sometime soon. On install users will be able to select from an alternate (not Google) DoH endpoint and accept the privacy policy for that service instead.

**Comment by jessejanderson** [comment]
I'm not sure this is worth the effort at this point? I want this to feel really easy for a new user without them needing to know or care too much about which provider is used for DNS or whatever.

Eventually, this will support multiple DID records and some sort of drop-down menu will likely be required. Perhaps at that time we could add a settings page to change this?

But I really want this extension to remain very low friction for people to use.

**Comment by sanitybit** [comment]
I agree with you; which is why I called it a boondoggle. I'm very much a privacy realist when it comes to using free web services, but I know there is a small but vocal subset of users with a different opinion.

I put this issue here as a signpost so that the small subset of privacy conscious users can see that it's a design choice that's at least been considered, and also to keep any conversation to a single issue.

You are correct in saying that the potential insertion into the onboarding workflow that I described above is too high friction, especially for non-technical users. I think that once an action popup to display multiple DIDs is implemented, a settings page with a selectable dropdown would be a better implementation. A settings page would also pave the way to supporting other configurable settings beyond the defaults (e.g. opening discovered profiles on different AT Protocol web frontends like `skyline.gay`).

Any future settings implementation would store a small list of DoH providers as consts, with the current setting saved in the extensions local storage. This would be similar to DoH server selection in Firefox settings.
